perfkitbenchmarker/analyze.py

```diff
@@ -19,13 +19,50 @@
 data format.
 """
 
-import pandas as pd
+import perfkitbenchmarker
 
-from perfkitbenchmarker import sample
 
+class Interval(object):
+  """Represents an interval in time.
 
-def AllStreamsInterval(start_times, durations, stream_ids):
-  """Compute when all streams were active from multistream records.
+  Args:
+    start: float. A POSIX timestamp.
+    duration: float. A length of time, in seconds.
+    end: float. A POSIX timestamp.
+
+  Either duration or end must be passed to the constructor.
+  """
+
+  def __init__(self, start, duration=None, end=None):
+    self.start = start
+
+    if duration is not None:
+      if end is not None and start + duration != end:
+        raise ValueError(
+            'Invalid arguments to interval constructor: %s + %s != %s' %
+            (start, duration, end))
+      self.duration = duration
+    elif end is not None:
+      self.duration = end - start
+
+
+  @property
+  def end(self):
+    return self.start + self.duration
+
+
+  def __eq__(self, other):
+    return (isinstance(other, Interval) and
+            self.start == other.start and
+            self.duration == other.duration)
+
+
+  def __repr__(self):
+    return 'Interval(%s, %s, %s)' % (self.start, self.duration, self.end)
+
+
+def AnyAndAllStreamsIntervals(start_times, durations, stream_ids):
```

Comment:
The name is cryptic. I start to understand it more when reading the comments.

How about simply: GetStreamActiveIntervals()? 

And then late the comments speak more for it.